trio/_core/tests/test_windows.py

```diff
@@ -1,13 +1,16 @@
 import os
+from threading import Thread
```

Comment:
I've [learned the hard way](https://github.com/python-trio/trio/pull/214) that it's usually a bad idea to use `threading.Thread` directly in tests – it can do things like swallow exceptions :-). If you need threads, then Trio's primitives are actually easier to use and more reliable, so ... that's what I use now.

Reply:
This tests a synchronous function, so afaik I cannot use a Trio primitive here, right? (I am not that familiar with Trio yet though :)). 

I now changed the test, so that the main thread waits for the handle, and a thread is spawned only to set the handle after a short timeout (there is thus very little that can go wrong in that thread).

Reply:
You can use an async test function to test a synchronous function :-).

```python
async def test_...():
    async with trio.open_nursery() as nursery:
        # This runs 'sync_fn' in a background thread
        nursery.start_soon(trio.run_sync_in_worker_thread, sync_fn)
        # ... we can do other stuff here while it's running ...
    # When we de-dent to close the nursery, it automatically joins the background thread
```

So this automatically makes sure that you join the background thread (it can't accidentally keep running while other tests go, even if this test crashes), it automatically detects exceptions in the background thread and makes sure that if they happen it causes the test to fail, and it's actually simpler to use (IMO) than the raw `threading.Thread` API :-).

Reply:
Of course. I don't understand why I did not think to use `run_sync_in_worker_thread` 🤔

Reply:
Hey, don't feel bad, this concurrency/IO stuff is complicated. I didn't think of it either when I was writing the `trio.ssl` testsuite, even though I *wrote* `run_sync_in_worker_thread` like the month before... so you get to learn from my fail :-)